### DISCUSSION TITLE
Add API documentation linter to CI

### DISCUSSION THREAD

**Discussion by nlohmann** [nlohmann/json#3624] [Ideas]
File `docs/mkdocs/scripts/check_structure.py` lints the API documentation. It currently is run manually, but could be overworked to fail on major issues.

**Comment by falbrechtskirchinger** [comment]
I think we should add NOLINT (maybe via HTML comments + a mkdocs extension for stripping comments). I'm addressing some of the style issues in #3553 which makes it easier to spot new ones but to fail on issues, we need the ability to silence false positives.

Then we can even add the style check to CI.
